contracts/custom/DecentDAOVoting.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity =0.8.19;
+
+import { LinearERC20Voting } from "../azorius/LinearERC20Voting.sol";
+import { VotesERC20Wrapper } from "../VotesERC20Wrapper.sol";
+
+interface Escrow {
+    function getBalance(address _address, uint _block) external;
+}
+
+contract DecentDAOVoting is LinearERC20Voting {
+
+    Escrow public escrow;
+
+    // TODO override setUp() function to initialize the escrow contract
+
+    function getVotingWeight(address _voter, uint32 _proposalId) public view override returns (uint256) {
+        return super.getVotingWeight(_voter, _proposalId) + escrow.getBalance(_voter, proposalVotes[_proposalId].votingStartBlock);
```

Comment:
@tbwebb22 @adamgall does this make sense?  Basically all we should need to do is add the address's true voting weight to the escrow's balance.

This has the issue that it allows users to delegate what they hold themselves, but not sure if that's a problem or not.

Reply:
> This has the issue that it allows users to delegate what they hold themselves, but not sure if that's a problem or not.

To be clear, is the issue that it allows users to delegate what they hold themselves _but not_ delegate their balance held in the lock contract? So there's like a partial delegation happening?

Reply:
yeah, just flagging that that's a possibility

folks would always have a voting weight equal to their locked tokens, but would be able to either self delegate or delegate to someone else any tokens they hold themselves

Reply:
just curious, is it possible to "turn off" delegation for unlocked tokens within this strategy? so that no delegation at all is happening?

Reply:
we call `getPastVotes` on the `IVotes` (https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/utils/IVotes.sol) contract, so the ERC20 token would probably need to extended and remove the delegation aspect of the snapshotting.

Reply:
@sethhrbek

Could you please explain more about point 3 of Keeping Delegation? From which point of view are you making that statement? If I own tokens, and delegate the voting power of those tokens to myself or someone else, and then transfer my tokens to another address, the voting power of those tokens does indeed change to whoever the new address has delegated their voting power to.

Also, apologies, but I'm struggling to understand both points in Removing Delegation. You don't need to talk more about it here in this thread, we can catch up next week about it.

Reply:
@sethhrbek @adamgall just pushed an example of how to prevent delegation while still maintaining the snapshotting functionality

Also as to point 1, if we did have delegation any vested tokens would be delegated to yourself as soon as they're unlocked, if you have yourself as the delegateee